Fix anticheat false positives when using fast with speed physics override

Desour opened the pull request:
Fixes  #14061.

You can also look at the changes of https://github.com/minetest/minetest/commit/8ebaf753d3c6b50028d097db924a887d98602c18 in `src/server/player_sao.cpp`. `player_max_walk` used to be `m_player->movement_speed_fast * m_player->physics_override.speed`.

## To do

This PR is a Ready for Review.

## How to test

* Disable `disable_anticheat`.
* Start a multiplayer server.
* `/lua me:set_physics_override({speed=2.5})` (With some fork of luacmd, e.g. <https://github.com/Desour/luacmd>.)
* Fly or walk around, with fast.

SmallJoker commented on src/server/player_sao.cpp:
```diff
@@ -618,7 +618,7 @@ bool PlayerSAO::checkMovementCheat()
 	float speed_walk = m_player->movement_speed_walk * m_player->physics_override.speed;
-	float speed_fast = m_player->movement_speed_fast;
+	float speed_fast = m_player->movement_speed_fast * m_player->physics_override.speed;
 	float speed_crouch = m_player->movement_speed_crouch * m_player->physics_override.speed_crouch;
```
https://github.com/minetest/minetest/blob/6106e4e72b5d337819b16d7aace52610d948131c/src/client/localplayer.cpp#L694-L697

All speeds are affected, including crouch. It might make sense to first figure out the maximal speed and then apply the speed modifier on top of that. Like it was done before the PR:
https://github.com/minetest/minetest/commit/8ebaf753d3c6b50#diff-60afc4b1ee298e01efae55db3b202d0bdbffd3919d7f0bc3212a4d300522eacaL616

Desour replied:
I've now applied the speed override where applicable: walk, crouch, fast, and climb

(I've decided against multiplying afterwards for consistency with max jump, in which case the speed override is not applied for the maximum of everything, but just for climb.)

Warr1024 commented:
Compiled and tested it with NodeCore and it seems to be working fine.  Tested it in devtest with speed=10 while walking, flying, and sneak-walking, and it seemed fine.

Albeit without knowing all the potential nuances, the code looks fine to me.

Desour commented:
Updated to include same fix for climbing and crouching.

Note: Flying and swimming is handled by the `In certain cases (swimming, climbing, flying) walking speed is applied vertically` thing.

Reproduction instructions also updated.

SmallJoker reviewed (approved):
Reproduced the position resets in master. This PR fix them.
